### ISSUE TITLE
`... with` provokes internal error

### ISSUE THREAD

**Issue by lawcho** [agda/agda#5890]
Removing the `...` before `with` makes the code pass type-checking.

The error:
```
An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE__, called at src/full/Agda/TypeChecking/Rules/Def.hs:479:8 in Agda-2.6.2.1-3iYJFSRK38DIOELxjGwiJC:Agda.TypeChecking.Rules.Def
```

The code:
```
open import Agda.Builtin.Unit

ok : ⊤          -- Passes as expected
ok
    with tt
... | x
    with tt
... | y = tt

bad : ⊤         -- Internal error
bad
    with tt
... | x
... with tt
... | y = tt
```

**Comment by gallais** [comment]
This is probably #5728

**Comment by lawcho** [comment]
Looks like it -- closing as duplicate.

**Comment by andreasabel** [comment]
Fixed along #5728.
